### PR TITLE
Make an initial attempt at very basic TypeScript

### PR DESCRIPTION
**Pull request by kylelaker**
This integrates the very beginnings of TypeScript into the codebase.
This is far from good. Nearly everything is currently typed as `any` and
the things that aren't may not have their types specified correctly.

This is a proof of concept and we may not actually use this at any point
in time ever. It just demonstrates that theoretically, incremental
adoption may be an option for us.

### CODE REVIEW

**Comment by kylelaker** [comment]
What we're really missing is a library that defines the OSCAL specification as TypeScript types. https://github.com/NREL-CYBER/oscal is super cool and will help with this, but it renames some fields (to use snake_case instead of kebab-case) and requires using some import/export functions defined within the library.

File: example/package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "eject": "node ../node_modules/react-scripts/bin/react-scripts.js eject"
   },
   "dependencies": {
-    "@EasyDynamics/oscal-react-library": "file:..",
+    "@easydynamics/oscal-react-library": "file:..",
```
Comment (tuckerzp) [inline example/package.json:18]: Is there a reason we switched from `EasyDynamics --> easydynamics`? 

I am wondering if that is what is causing the error:
```
Module not found: Error: Can't resolve '@easydynamics/oscal-react-library' in '/home/runner/work/oscal-react-library/oscal-react-library/example/src'
```

Comment (kylelaker) [reply example/package.json:18]: That was done as part of publishing to `npm`; we have to be all-lowercase. Make sure to `npm ci` in both directories.

File: src/components/oscal-utils/OSCALRestUtils.tsx [hunk 2]
```diff
@@ -241,12 +259,12 @@ export function createOrUpdateSspControlImplementationImplementedRequirementStat
   if (implementationSetParameters?.length) {
     statementByComponent["set-parameters"] ??= [];
```
Comment (tuckerzp) [inline src/components/oscal-utils/OSCALRestUtils.tsx:260]: On lines: `239`, `253`, & `260` there is the error:
```
/OSCAL/oscal-react-library/src/components/oscal-utils/OSCALRestUtils.tsx(239,52): syntax error TS1109: Expression expected.
```
Despite both babel & typescript supporting Logical nullish assignment in their current versions.

**Comment by kylelaker** [comment]
I think a lot of our issues here related back to #403

**Comment by tuckerzp** [comment]
I am attempting to use https://github.com/jaredpalmer/tsdx. I wanted to see how big of an effort it would be to switch to using that.

**Pull request closed without merging**